sceptre/cli/update.py

```diff
@@ -52,31 +52,46 @@ def update_command(ctx, path, change_set, verbose, yes):
     plan = SceptrePlan(context)
 
     if change_set:
+        delete_change_set = True
+
         change_set_name = "-".join(["change-set", uuid1().hex])
         plan.create_change_set(change_set_name)
         try:
             # Wait for change set to be created
             statuses = plan.wait_for_cs_completion(change_set_name)
             # Exit if change set fails to create
+            atleast_one_ready = False
             for status in list(statuses.values()):
-                if status != StackChangeSetStatus.READY:
+                if status != StackChangeSetStatus.READY \
+                        and status != StackChangeSetStatus.NO_CHANGES:
+                    # TODO: Print an informative message to the user.
                     exit(1)
+                if status == StackChangeSetStatus.READY:
+                    atleast_one_ready = True
+
+            if not atleast_one_ready:
+                write("No changes detected", context.output_format)
+                exit(0)
```

Comment:
I'm wondering why we wouldn't wait for ALL change sets to be in a "ready" / "no change" status. Why would we fire away when there's "at least one" while others could be failed or still running?

Reply:
That is effectively what this is doing. At this point we have a non-pending status for all change sets, and this merely indicates if at least one of them came back ready, as opposed to with "no changes". If any failed we would have exited already.